pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -157,10 +157,14 @@ func (c *Reconciler) reconcile(ctx context.Context, pa *pav1alpha1.PodAutoscaler
 	//			this revision, e.g. after a restart) but PA status is inactive (it was
 	//			already scaled to 0).
 	// 2. The excess burst capacity is negative.
-	if want == 0 || decider.Status.ExcessBurstCapacity < 0 || want == -1 && pa.Status.IsInactive() {
-		logger.Infof("SKS should be in proxy mode: want = %d, ebc = %d, PA Inactive? = %v",
-			want, decider.Status.ExcessBurstCapacity, pa.Status.IsInactive())
+	if (pa.Status.IsActivating() && want == 1) || want == 0 || decider.Status.ExcessBurstCapacity < 0 || want == -1 && pa.Status.IsInactive() {
```

Comment:
How does this change fit into the PR?

Reply:
Thank you for your review&comment. I had found that switching the endpoint of the public service causes 503, so I wanted to see if this change fixes the problem or not.
But now, I could figure out that it sems a pure Istio issue so opened https://github.com/istio/istio/issues/17837 with the reproducer.